display_list/display_list_paint.h

```diff
@@ -0,0 +1,218 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_DISPLAY_LIST_PAINT_H_
+#define FLUTTER_DISPLAY_LIST_DISPLAY_LIST_PAINT_H_
+
+#include "flutter/display_list/display_list_blend_mode.h"
+#include "flutter/display_list/display_list_color_filter.h"
+#include "flutter/display_list/display_list_color_source.h"
+#include "flutter/display_list/display_list_image_filter.h"
+#include "flutter/display_list/display_list_mask_filter.h"
+
+namespace flutter {
+
+class DlColor {
+ public:
+  DlColor() : argb(0xFF000000) {}
+  DlColor(uint32_t argb) : argb(argb) {}
+
+  uint32_t argb;
+
+  bool operator==(DlColor const& other) const { return argb == other.argb; }
+  bool operator!=(DlColor const& other) const { return argb != other.argb; }
+};
```

Comment:
nit: any reason to not just have this be a typedef?

Reply:
A couple of nit-ty reasons.

- SkColor being a typedef means you can't overload off of it.
- Other type-safety-ish consequences
- (In my "verbose DL compare" code, for example, I had to wrap all SkColors in a class to get them to print easily)
- Potentially these can contain a 4-float version, but I haven't decided how to manage that distinction yet.
- Starting with a (simple, non-virtual) class will demonstrate if there are any negative consequences as we move forward with this